src/puppetlabs/puppetdb/http/events.clj

```diff
@@ -6,40 +6,41 @@
             [puppetlabs.puppetdb.middleware :as middleware]
             [net.cgrand.moustache :refer [app]]
             [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.schema :as pls :refer [defn-validated]]
+            [schema.core :as s]
             [puppetlabs.puppetdb.time :refer [to-timestamp]]))
 
-(defn validate-distinct-options!
+(defn-validated validate-distinct-options! :- {:distinct_resources? s/Bool
+                                               :distinct_start_time (s/maybe pls/Timestamp)
+                                               :distinct_end_time (s/maybe pls/Timestamp)}
   "Validate the HTTP query params related to a `distinct_resources` query.  Return a
   map containing the validated `distinct_resources` options, parsed to the correct
   data types.  Throws `IllegalArgumentException` if any arguments are missing
   or invalid."
-  [params]
-  {:pre [(map? params)]
-   :post [(map? %)
-          (every? (partial contains? %) #{:distinct_resources? :distinct_start_time :distinct_end_time})
-          (kitchensink/boolean? (:distinct_resources? %))
-          ((some-fn (partial instance? Timestamp) nil?) (:distinct_start_time %))
-          ((some-fn (partial instance? Timestamp) nil?) (:distinct_end_time %))]}
-  (let [distinct-params ["distinct_resources" "distinct_start_time" "distinct_end_time"]]
-    (cond
-     (not-any? #(contains? params %) distinct-params)
+  [params :- {s/Any s/Any}]
+  (let [distinct-params-names ["distinct_resources" "distinct_start_time" "distinct_end_time"]
+        {:strs [distinct_start_time distinct_end_time] :as distinct-params}
+        (select-keys params distinct-params-names)]
+    (condp = (kitchensink/keyset distinct-params)
+     #{}
      {:distinct_resources? false
       :distinct_start_time nil
       :distinct_end_time   nil}
 
-     (every? #(contains? params %) distinct-params)
-     (let [start (to-timestamp (params "distinct_start_time"))
-           end   (to-timestamp (params "distinct_end_time"))]
+     (set distinct-params-names)
```

Comment:
This one gets used in the select-keys above,not sure if that takes a set arg instead of a vec